src/api/artifact/controller.go

```diff
@@ -516,6 +521,16 @@ func (c *controller) assembleArtifact(ctx context.Context, art *artifact.Artifac
 	artifact := &Artifact{
 		Artifact: *art,
 	}
+
+	if artifact.RepositoryName == "" {
```

Comment:
I'd prefer to store the repository name into DB.

Reply:
Agree. I opened an issue to track the work https://github.com/goharbor/harbor/issues/10805